src/OpenTK/Math/Vector3.cs

```diff
@@ -554,6 +584,28 @@ public static void Clamp(ref Vector3 vec, ref Vector3 min, ref Vector3 max, out
         }
 
         /// <summary>
+        /// Compute the euclidean distance between two vectors.
+        /// </summary>
+        /// <param name="vec1">The first vector</param>
+        /// <param name="vec2">The second vector</param>
+        /// <returns>The distance</returns>
+        public static float Distance(Vector3 vec1, Vector3 vec2)
+        {
+            return (vec1 - vec2).Length;
```

Comment:
This could be optimized to avoid allocating the intermediate vector (on which length) is then called.

We also very likely want to provide a 'ref' version to avoid copying both vectors. You can see how most of ther other methods do this.